netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -247,9 +254,18 @@ private NettyTransportFactory(Class<? extends Channel> channelType,
 
     @Override
     public ClientTransport newClientTransport(SocketAddress serverAddress, String authority) {
-      GrpcUtil.checkAuthority(authority);
-      return new NettyClientTransport(serverAddress, channelType, group,
-          createProtocolNegotiator(authority, negotiationType, sslContext),
+      String host = "";
+      int port = -1;
+      // Don't check the authority again here, because it will have already been validated by the
+      // parent class.
+      if (serverAddress instanceof InetSocketAddress) {
```

Comment:
I don't think this is right. Any solution that involves looking at the InetSocketAddress to choose the host is probably wrong (at this point in the code; within the ManagedChannelBuilder can be fine).

The authority has already been chosen at this point and does not have to match the InetSocketAddress. Even if the authority can't be parsed into host/port we still wouldn't want to use the socket address since the address with high probability doesn't correspond to the same thing as the authority (as otherwise, the authority would have been in the host:port form).